src/node/rpc/claims.h

```diff
@@ -14,14 +14,31 @@ namespace ccf
 
   static crypto::Sha256Hash entry_leaf(
     const std::vector<uint8_t>& write_set,
-    const ClaimsDigest::Digest& claims_digest)
+    const ClaimsDigest::Digest& claims_digest,
+    const crypto::Sha256Hash& commit_evidence_digest)
   {
     crypto::Sha256Hash write_set_digest({write_set.data(), write_set.size()});
     LOG_TRACE_FMT(
-      "entry_leaf {} + {} = {}",
+      "entry_leaf ws: {} + cd: {} + ce: {} = {}",
       write_set_digest,
       claims_digest,
-      crypto::Sha256Hash(write_set_digest, claims_digest));
-    return crypto::Sha256Hash(write_set_digest, claims_digest);
+      commit_evidence_digest,
+      crypto::Sha256Hash(
+        write_set_digest, claims_digest, commit_evidence_digest));
+    return crypto::Sha256Hash(
+      write_set_digest, claims_digest, commit_evidence_digest);
+  }
+
+  static crypto::Sha256Hash entry_leaf(
+    const std::vector<uint8_t>& write_set,
+    const crypto::Sha256Hash& commit_evidence_digest)
```

Comment:
Would it be simpler to have only one `entry_leaf()` function with an optional `claims_digest` argument?

Reply:
Yes, but then it means either a ref to an optional, which we have to construct when there's no claims, or switching to a ptr. Meh?

Reply:
`const std::optional<ClaimsDigest::Digest>& claims_digest = std::nullopt` should be OK, no?

Reply:
Sure but then we have to construct a std::nullopt even when there are no claims.

Reply:
Not if `claims_digest` is the last argument to the function. See e.g. https://github.com/microsoft/CCF/blob/f6f4bcecd9e397ef69fc14780d5b6892a1f961bc/src/consensus/aft/raft.h#L1429

Reply:
I'm not convinced this buys us meaningful simplicity, we only get to de-duplicate one line: the write set digest computation, only to dispatch to the two paths in a runtime branch.